app/models/dialog/container_template_service_dialog.rb

```diff
@@ -74,7 +74,7 @@ def add_parameter_field(param, group, position)
         :name          => "param_#{param.name}",
         :data_type     => "string",
         :display       => "edit",
-        :required      => param.required,
+        :required      => param.required && param.generate.nil?,
```

Comment:
@lfu Overall this looks good.  Just wondering if `param.gererate` could ever be an empty string.  Suggest using `param.generate.blank?` unless you know or it is documented to always be `nil` in this case.